src/ember-app.js

```diff
@@ -210,6 +222,19 @@ class EmberApp {
       fastbootInfo: fastbootInfo
     });
 
+    let destroyAppInstanceTimer;
+    if (destroyAppInstanceInMs) {
+      // start a timer to destroy the appInstance forcefully in the given ms.
+      // This is a failure mechanism so that node process doesn't get wedged if the `visit` never completes.
+      destroyAppInstanceTimer = setTimeout(function() {
```

Comment:
On other way probably is to do a `Ember.run.debounce`  so that it is hooked on to Ember's run loop. I am open to using that at both places if needed but it requires some more refactoring to run this in the sandbox context.